From 58cc992652d53444ce22c6759afb91c810530558 Mon Sep 17 00:00:00 2001
From: YouMin Chen <cym@rock-chips.com>
Date: Mon, 8 Jun 2020 20:03:44 +0800
Subject: [PATCH] clk: rockchip: clk-ddr: fix return value in
 rockchip_ddrclk_sip_set_rate

Returns success (0) or negative errno in rockchip_ddrclk_sip_set_rate call.

Change-Id: I9c424d8625a465c235ac5f6b0795f51848e65283
Signed-off-by: YouMin Chen <cym@rock-chips.com>
---
 drivers/clk/rockchip/clk-ddr.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/clk/rockchip/clk-ddr.c b/drivers/clk/rockchip/clk-ddr.c
index c2cb6c937cd4..41877f5177c3 100644
--- a/drivers/clk/rockchip/clk-ddr.c
+++ b/drivers/clk/rockchip/clk-ddr.c
@@ -102,7 +102,10 @@ static int rockchip_ddrclk_sip_set_rate(struct clk_hw *hw, unsigned long drate,
 		      ROCKCHIP_SIP_CONFIG_DRAM_SET_RATE,
 		      0, 0, 0, 0, &res);
 
-	return res.a0;
+	if (res.a0)
+		return 0;
+	else
+		return -EPERM;
 }
 
 static unsigned long
-- 
2.35.3

